analyses/summary/helper_test.go

```diff
@@ -0,0 +1,68 @@
+package summary_test
+
+import (
+	"testing"
+
+	"github.com/fabric8-analytics/cli-tools/analyses/driver"
+	"github.com/fabric8-analytics/cli-tools/analyses/summary"
+)
+
+func data() driver.GetResponseType {
+	var GetResponse = &driver.GetResponseType{
+		AnalysedDeps: []driver.AnalysedDepsType{
+			{
+				Ecosystem:     "pypi",
+				LatestVersion: "7.1.2",
+				Name:          "click",
+				PublicVulnerabilities: []driver.VulnerabilitiesType{
+					{
+						Severity: "medium",
+						ID:       "ABC-PYTHON-CODECOV-12345",
+						Title:    "Command Injection",
+					},
+				},
+			},
+		},
+	}
+	return *GetResponse
+}
+
+func TestGetResultSummary(t *testing.T) {
+	got := summary.GetResultSummary(data())
+	want := &summary.StackSummary{
+		TotalScannedDependencies:           1,
+		TotalScannedTransitiveDependencies: 0,
+		TotalVulnerabilities:               0,
+		CommonlyKnownVulnerabilities:       0,
+		VulnerabilitiesUniqueToSynk:        0,
+		DirectVulnerableDependencies:       1,
+		CriticalVulnerabilities:            0,
+		HighVulnerabilities:                0,
+		MediumVulnerabilities:              1,
+		LowVulnerabilities:                 0,
+	}
+	if want.CriticalVulnerabilities != got.CriticalVulnerabilities {
+		t.Errorf("Invalid CriticalSeverity, got: %v, want: %v", got.CriticalVulnerabilities, want.CriticalVulnerabilities)
+	}
+	if want.HighVulnerabilities != got.HighVulnerabilities {
+		t.Errorf("Invalid HighVulnerabilities, got: %v, want: %v", got.HighVulnerabilities, want.HighVulnerabilities)
+	}
+	if want.MediumVulnerabilities != got.MediumVulnerabilities {
+		t.Errorf("Invalid MediumVulnerabilities, got: %v, want: %v", got.MediumVulnerabilities, want.MediumVulnerabilities)
+	}
+	if want.LowVulnerabilities != got.LowVulnerabilities {
+		t.Errorf("Invalid LowVulnerabilities, got: %v, want: %v", got.LowVulnerabilities, want.LowVulnerabilities)
+	}
+	if want.TotalScannedDependencies != got.TotalScannedDependencies {
+		t.Errorf("Invalid TotalScannedDependencies,: %v, want: %v", got, want)
+	}
+	if want.VulnerabilitiesUniqueToSynk != got.VulnerabilitiesUniqueToSynk {
+		t.Errorf("Invalid VulnerabilitiesUniqueToSynk, got: %v, want: %v", got, want)
+	}
+	if want.DirectVulnerableDependencies != got.DirectVulnerableDependencies {
+		t.Errorf("Invalid DirectDependenciesWithKnownVulnerabilities, got: %v, want: %v", got, want)
+	}
+	if want.TotalScannedDependencies != got.TotalScannedDependencies {
+		t.Errorf("Invalid TotalScannedDependencies, got: %v, want: %v", got, want)
+	}
```

Comment:
Use `github.com/google/go-cmp/cmp` to compare two structs directly.
```suggestion
	if diff := cmp.Diff(want, got); diff != "" {
	        t.Errorf("Vuln mismatch (-want, +got):\n%s", diff)
	}
```

Reply:
Ack.